frontend/src/pages/projects/notebook/DeleteNotebookModal.tsx

```diff
@@ -36,10 +43,18 @@ const DeleteNotebookModal: React.FC<DeleteNotebookModalProps> = ({ notebook, onC
           (envFrom) => !envFrom.secretRef?.name.includes(DATA_CONNECTION_PREFIX),
         );
         const configMapNames = nonDataConnectionVariables
-          .filter((envName): envName is ConfigMapRef => !!envName.configMapRef)
+          .filter(
+            (envName): envName is ConfigMapRef =>
+              !!envName.configMapRef &&
+              Boolean(envName.configMapRef.name.match(new RegExp(`^${CONFIGMAP_PREFIX}.{6}$`))),
```

Comment:
`.{6}` in regexp can be any 6 characters... they can be all dashes but the last character and still be valid k8s name but not what we are looking for.

The DSPA secret looks for `[a-z0-9]{6}` -- I think that is more accurate of a check.